communication/src/TlsTCPCommunication.cpp

```diff
@@ -910,8 +910,15 @@ class AsyncTlsConnection : public std::enable_shared_from_this<AsyncTlsConnectio
    * @param length data length
    */
   void send(const char *data, uint32_t length) {
-    assert(data);
-    assert(length > 0 && length <= _maxMessageLength - MSG_HEADER_SIZE);
+    if (!data) {
+      LOG_ERROR(_logger, "Error, message has never been initialized")
+      throw std::runtime_error("data = nullptr");
```

Comment:
Both conditions check input parameters. Using `std::invalid_argument` might be better suited?!

Reply:
std::invalid_argument seems suitable, thanks!